### PR TITLE
Move `sqlite/*.go` to `internal/handler`

### PR DESCRIPTION
**Pull request by rumyantseva**
# Description

I updated everything I found relevant. Please let me know if you see any other places that need to be updated in the current PR.

Closes #3662.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [x] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (rumyantseva) [inline .github/CODEOWNERS:12]: I kept this to make it explicit that hana backend has different codeowners rules (even after we set codeowners to the team). Alternatively, I can remove this line completely.

Comment (AlekSi) [reply .github/CODEOWNERS:12]: It is actually not possible to add them there, see (private) FerretDB/ferreteria#45

File: internal/handler/handlers_test.go
Comment (AlekSi) [inline internal/handler/handlers_test.go:1]: This file should be renamed too

**Review by noisersup — DISMISSED** [review]
apart from above no changes needed! LGTM

**Review by noisersup — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Pull request merged**
